### DISCUSSION TITLE
May I ask how to find and register an email based on the username,

### DISCUSSION THREAD

**Discussion by Chineseweisheng** [community/community#123775] [New to GitHub]
May I ask how to find the registered email address based on the username? I forgot my password, and I have too many email addresses. I don't know which email registered the account for. If I find the email address, I can apply to change my password through email.

**Comment by nikhilsbuilds** [comment]
Unfortunately, GitHub does not provide a direct way to find the registered email address associated with a username. The email address used to register a GitHub account is private information that is not publicly visible.

However, there are a few steps you can take to try and recover your GitHub account:

1. **Password Reset**: On the GitHub login page, click on the "Forgot password?" link. This will prompt you to enter the email address associated with your account. GitHub will then send a password reset link to that email address.

2. **Email Addresses on File**: If you have access to any of the email accounts you've used in the past, check the inbox and spam/junk folders for emails from GitHub. These emails may contain clues about which address is associated with your account.

3. **GitHub Support**: If you've exhausted the previous options and still can't remember the email address, you can contact GitHub Support for assistance. They may be able to help you recover your account after verifying your identity and ownership of the account.

It's generally a good practice to use a personal email address that you have full control over when creating accounts on websites and services. This can help avoid situations where you lose access to an account due to forgetting the associated email address.
